packages/react/src/components/radio-group/radio-group.tsx

```diff
@@ -1,35 +1,65 @@
-import { createRadioGroupStore, RADIO_GROUP_ACTIONS } from '@devextreme/components';
+/* eslint-disable react/jsx-props-no-spreading */
 import {
-  Children, cloneElement, isValidElement, memo, useMemo,
+  createContainerPropsSelector,
+  createRadioGroupStore,
+  RADIO_GROUP_ACTIONS,
+  RADIO_GROUP_CONTAINER_PROPS_BUILDER,
+} from '@devextreme/components';
+import {
+  Children,
+  cloneElement,
+  ForwardedRef,
+  forwardRef,
+  isValidElement,
+  memo,
+  PropsWithChildren,
+  useMemo,
 } from 'react';
-import { useCallbackRef, useSecondEffect } from '../../internal/hooks';
-import { EditorProps } from '../../internal/props';
+import { useCallbackRef, useSecondEffect, useStoreSelector } from '../../internal/hooks';
+import { EditorProps, FocusableComponent } from '../../internal/props';
 import { RadioGroupStoreContext } from '../radio-common';
 
-function RadioGroupInternal<T>(props: RadioGroupProps<T>) {
-  const controlledMode = useMemo(() => Object.hasOwnProperty.call(props, 'value'), []);
+import '@devextreme/styles/src/radio-group/radio-group.scss';
+
+function RadioGroupInternal<T>(
+  props: RadioGroupProps<T>,
+  ref: ForwardedRef<HTMLDivElement>,
+) {
+  const isValueControlled = useMemo(() => Object.hasOwnProperty.call(props, 'value'), []);
   const valueChange = useCallbackRef(props.valueChange);
 
   const store = useMemo(() => createRadioGroupStore<T>({
-    value: controlledMode ? props.value : props.defaultValue,
+    readonly: RADIO_GROUP_CONTAINER_PROPS_BUILDER.getDomOptions(props),
+    value: isValueControlled ? props.value : props.defaultValue,
   }, {
     value: {
-      controlledMode,
+      controlledMode: isValueControlled,
       changeCallback: (value: T) => { valueChange.current(value); },
     },
   }), []);
 
+  const containerProps = useStoreSelector(store, createContainerPropsSelector, []);
+
+  const readonlyValues = RADIO_GROUP_CONTAINER_PROPS_BUILDER.getDomOptions(props);
   useSecondEffect(() => {
-    if (controlledMode) {
+    if (isValueControlled) {
       store.addUpdate(RADIO_GROUP_ACTIONS.updateValue(props.value));
     }
 
+    store.addUpdate(RADIO_GROUP_ACTIONS.updateReadonly(readonlyValues));
+
     store.commitPropsUpdates();
-  }, [props.value]);
+  }, [props.value, ...Object.values(readonlyValues)]);
```

Comment:
Question: Do we really need useEffect here? May be it's a premature optimization? We have already have memozed selector that can solve problem of unnecessary mutations and already now the deps between state and ViewModel. Why we dublication knolodge about deps here?

Reply:
Yes, we really need ```useEffect``` here :)